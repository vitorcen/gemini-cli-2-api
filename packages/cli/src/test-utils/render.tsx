/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import { render } from 'ink-testing-library';
import type React from 'react';
import { LoadedSettings } from '../config/settings.js';
import { KeypressProvider } from '../ui/contexts/KeypressContext.js';
import { SettingsContext } from '../ui/contexts/SettingsContext.js';
import { ShellFocusContext } from '../ui/contexts/ShellFocusContext.js';

const mockSettings = new LoadedSettings(
  { path: '', settings: {}, originalSettings: {} },
  { path: '', settings: {}, originalSettings: {} },
  { path: '', settings: {}, originalSettings: {} },
  { path: '', settings: {}, originalSettings: {} },
  true,
  new Set(),
);

export const renderWithProviders = (
  component: React.ReactElement,
  { shellFocus = true, settings = mockSettings } = {},
): ReturnType<typeof render> =>
  render(
    <SettingsContext.Provider value={settings}>
      <ShellFocusContext.Provider value={shellFocus}>
        <KeypressProvider kittyProtocolEnabled={true}>
          {component}
        </KeypressProvider>
      </ShellFocusContext.Provider>
    </SettingsContext.Provider>,
  );
