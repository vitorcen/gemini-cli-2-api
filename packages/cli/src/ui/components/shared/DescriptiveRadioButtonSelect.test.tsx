/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import { describe, it, expect, vi, beforeEach } from 'vitest';
import { renderWithProviders } from '../../../test-utils/render.js';
import {
  DescriptiveRadioButtonSelect,
  type DescriptiveRadioSelectItem,
  type DescriptiveRadioButtonSelectProps,
} from './DescriptiveRadioButtonSelect.js';

vi.mock('./BaseSelectionList.js', async (importOriginal) => {
  const actual =
    await importOriginal<typeof import('./BaseSelectionList.js')>();
  return {
    ...actual,
    BaseSelectionList: vi.fn(({ children, ...props }) => (
      <actual.BaseSelectionList {...props}>{children}</actual.BaseSelectionList>
    )),
  };
});

vi.mock('../../semantic-colors.js', () => ({
  theme: {
    text: {
      primary: 'COLOR_PRIMARY',
      secondary: 'COLOR_SECONDARY',
    },
    status: {
      success: 'COLOR_SUCCESS',
    },
  },
}));

describe('DescriptiveRadioButtonSelect', () => {
  const mockOnSelect = vi.fn();
  const mockOnHighlight = vi.fn();

  const ITEMS: Array<DescriptiveRadioSelectItem<string>> = [
    {
      title: 'Foo Title',
      description: 'This is Foo.',
      value: 'foo',
      key: 'foo',
    },
    {
      title: 'Bar Title',
      description: 'This is Bar.',
      value: 'bar',
      key: 'bar',
    },
    {
      title: 'Baz Title',
      description: 'This is Baz.',
      value: 'baz',
      disabled: true,
      key: 'baz',
    },
  ];

  const renderComponent = (
    props: Partial<DescriptiveRadioButtonSelectProps<string>> = {},
  ) => {
    const defaultProps: DescriptiveRadioButtonSelectProps<string> = {
      items: ITEMS,
      onSelect: mockOnSelect,
      ...props,
    };
    return renderWithProviders(
      <DescriptiveRadioButtonSelect {...defaultProps} />,
    );
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('should render correctly with default props', () => {
    const { lastFrame } = renderComponent();
    expect(lastFrame()).toMatchSnapshot();
  });

  it('should render correctly with custom props', () => {
    const { lastFrame } = renderComponent({
      initialIndex: 1,
      isFocused: false,
      showScrollArrows: true,
      maxItemsToShow: 5,
      showNumbers: true,
      onHighlight: mockOnHighlight,
    });
    expect(lastFrame()).toMatchSnapshot();
  });
});
