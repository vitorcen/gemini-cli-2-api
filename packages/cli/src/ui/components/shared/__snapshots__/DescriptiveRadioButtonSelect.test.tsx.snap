// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DescriptiveRadioButtonSelect > should render correctly with custom props 1`] = `
"▲
  1. Foo Title
     This is Foo.
● 2. Bar Title
     This is Bar.
  3. Baz Title
     This is Baz.
▼"
`;

exports[`DescriptiveRadioButtonSelect > should render correctly with default props 1`] = `
"● Foo Title
  This is Foo.
  Bar Title
  This is Bar.
  Baz Title
  This is Baz."
`;
